src/app/core/services/notifications/fcm-xmpp-notification.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from '@angular/core'
+import { Firebase } from '@ionic-native/firebase/ngx'
+import { Platform } from 'ionic-angular'
+import * as uuid from 'uuid/v4'
+
+import { DefaultMaxUpstreamResends } from '../../../../assets/data/defaultConfig'
+import { SingleNotification } from '../../../shared/models/notification-handler'
+import { TaskType } from '../../../shared/utilities/task-type'
+import { getSeconds } from '../../../shared/utilities/time'
+import { RemoteConfigService } from '../config/remote-config.service'
+import { SubjectConfigService } from '../config/subject-config.service'
+import { LocalizationService } from '../misc/localization.service'
+import { LogService } from '../misc/log.service'
+import { ScheduleService } from '../schedule/schedule.service'
+import { StorageService } from '../storage/storage.service'
+import { FcmNotificationService } from './fcm-notification.service'
+import { NotificationGeneratorService } from './notification-generator.service'
+
+declare var FirebasePlugin
+
+@Injectable()
+export class FcmXmppNotificationService extends FcmNotificationService {
+  constructor(
+    public notifications: NotificationGeneratorService,
+    public storage: StorageService,
+    public schedule: ScheduleService,
+    public config: SubjectConfigService,
+    public firebase: Firebase,
+    public platform: Platform,
+    public logger: LogService,
+    public remoteConfig: RemoteConfigService,
+    public localization: LocalizationService
+  ) {
+    super(storage, config, firebase, platform, logger, remoteConfig)
+  }
+
+  getSubjectDetails() {
+    return this.config.getParticipantLogin().then(subjectId => {
+      subjectId
+    })
+  }
```

Comment:
where is this used?

Reply:
In FcmNotificationService:
https://github.com/RADAR-base/RADAR-Questionnaire/blob/21ce57b3b65d47ccfa641a165862ba8df73ef772/src/app/core/services/notifications/fcm-notification.service.ts#L70-L90

Reply:
Ok, thanks. I think then it's better to add to this class. And because fcm-xmpp is a sub-class of fcm, it should be available in fcm-xmpp and fcm-rest too if we need it in the future